scripts/build_canonical_gliner_dataset.py

```
@@ -0,0 +1,836 @@
+        return None
+
+    return {
+        "tokenized_text": tokens,
+        "ner": [],
+        "ner_negatives": canonical_labels,
+        "is_negative": True,
+    }
+
+
+def choose_hash_split(source: str, text: str) -> str:
+    digest = hashlib.sha1(f"{source}\n{text}".encode("utf-8")).hexdigest()
```

Ruff (0.15.1)

[error] Probable use of insecure hash functions in `hashlib`: `sha1`

(S324)